### DISCUSSION TITLE
Copyright issues

### DISCUSSION THREAD

**Discussion by nalexn** [izzyumkin/EmojiPicker#20] [General] [answered]
Hey @izyumkin and @htmlprogrammist

Just wanted to point out that one of you may need to reference the other project, as yours [MCEmojiPicker](https://github.com/izyumkin/MCEmojiPicker) and [EmojiPicker](https://github.com/htmlprogrammist/EmojiPicker) seem to be almost identical (the readmes are 99% identical word-by-word).

**Comment by rxkesh** [comment]
Lol that is quite the copy paste, good eye!

**Comment by izyumkin** [comment]
Hey! Yes, it's very funny! Finally someone noticed it. You can look in the commit history [EmojiPicker](https://github.com/htmlprogrammist/EmojiPicker) who and when really made this library.

In addition, I wrote an article in several languages about how I developed this library, and this person copied it and also participated in competitions with it...

If you know what I can do with this, I would be very happy!

**Comment by htmlprogrammist** [comment]
Hello, guys,

In this whole story, @izyumkin misses the fact that he had the opportunity to develop together, no one "just copied" this project, a pull request was opened so that all developers could develop an open source library and an MIT license together – https://github.com/izyumkin/MCEmojiPicker/pull/7  

He could also tell me that he would not like to output this project as a package, then we would just close the pull request, delete everything and that's it. But instead he decided to copy everything from my work without any modification. Let’s look through the readme, for example, as @nalexn pointed out:

Mine updates and others: https://github.com/htmlprogrammist/EmojiPicker/commit/7592f8ab5b2c8b516aedb474d3cd0a353ec40d53#diff-b335630551682c19a781afebcf4d07bf978fb1f8ac04c6bf87428ed5106870f5 

https://github.com/htmlprogrammist/EmojiPicker/commits/main/README.md 

His updates
https://github.com/izyumkin/MCEmojiPicker/commit/ad839cbdddfb6ddb09dfd2936bd0b3c60b05d983

https://github.com/izyumkin/MCEmojiPicker/commit/e1219256cebb3aabdbae41703611e3f50abf73dc

https://github.com/izyumkin/MCEmojiPicker/commit/ea01387496ff188c9cae9cda47810767fcfdb0b4 (check commit comments here. And date. Immediately after my comment, he closed the pull request)
But from the history of his repository I can’t get this commit (maybe, you too?). The only commit is https://github.com/izyumkin/MCEmojiPicker/commit/e47e3c005334395b94f869beaf7ccd4556b6bfa9 

Flags in localization (check dates)
https://github.com/izyumkin/MCEmojiPicker/commit/6725395537c061f3c4d10c133a390a7eff4ecbf0 
https://github.com/htmlprogrammist/EmojiPicker/tree/de2d4d7288baaa51f41cce1cda1f4890329b4726 

The author of the written code was admitted since the first commit of working on this project https://github.com/htmlprogrammist/EmojiPicker/commit/80990c2d79a13c015375676857cab8ae22220800#diff-94c67287601e99038cfdf11ed6bea8cb2edacea630b249a3b42e44bc0fadd67b and then I opened pull request

**Comment by nalexn** [comment, marked as answer]
@htmlprogrammist you absolutely have the right to fork a project and continue the development independently - open source community works this way. Just as @izyumkin has the right to not accept PRs. [MIT licence](https://www.tldrlegal.com/license/mit-license) just requires the project that appeared second to keep the copyright notice in the existing files of the project and leave a link to the original. Just mention it was inspired by that library, but now is developed independently

**Reply by htmlprogrammist** [reply]
Thanks for your reply. 

My claim was only whether @izyumkin had the right to copy everything that was written without mentioning me. Someone may find it wrong, someone may find it normal, because everything is under the MIT License, fortunately this story works both ways. 

As a person with impeccable karma, I will agree to mark it as you said, because I have nothing to hide and everything is confirmed by the commit dates.

**Comment by izyumkin** [comment]
@htmlprogrammist , yes, indeed our approaches are slightly different and that's why I closed your PR. But undoubtedly you have written good documentation for my library. At that time, I didn't know how to accept only one file from the pull request, and therefore I added the changes separately myself.

If you want, you can make a new pull request with some changes in the readme file to "stay in the commit history" or suggest how I can still add your changes to the readme from a closed PR. What do you think about this?

**Comment by htmlprogrammist** [comment]
@izyumkin To do what is fair, now I see options like this: reopen PR, accept changes, resolve all conflicts in your favor. But in this case maybe (maybe not) we will have to fix something later, for example, adding some now superfluous file. 

And the second option is simpler: mention me as the author of the documentation in the "About" of readme.md, for example, "Thanks to @htmlprogrammist for writing the documentation"

**Reply by izyumkin** [reply]
I think thanking only you in the library readme would be unfair to other contributors. So, I made a new branch in which I made a cherry pick of your commit where you wrote the documentation. Here's the [pull-request](https://github.com/izyumkin/MCEmojiPicker/pull/21). What do you think about this?

**Reply by htmlprogrammist** [reply]
Well, that's quite an idea. But if you chose this way, then cherry pick all commits related to this

**Reply by izyumkin** [reply]
For what? 

You want me to copyright the documentation I took from your repository, don't you?

This commit contains all the changes I took from your repository that have been saved so far. All the other changes you made I ended up replacing. I'd be happy to add them to their place in the commit history as well, but I don't think that's possible anymore.

**Reply by htmlprogrammist** [reply]
Everything is very simple. Just because it's not enough. 

> but I don't think that's possible anymore.

You can use the same method that you did earlier when you opened the pull request.

It was an easy search, it took me only 5-10 minutes, I think you can easily find more: 

Mine: https://github.com/htmlprogrammist/EmojiPicker/commit/7592f8ab5b2c8b516aedb474d3cd0a353ec40d53, https://github.com/htmlprogrammist/EmojiPicker/commit/ab8def5ea68391ae0aebe9771fb0c2bdc5a7759c
Yours: https://github.com/izyumkin/MCEmojiPicker/commit/ea01387496ff188c9cae9cda47810767fcfdb0b4

Mine: https://github.com/htmlprogrammist/EmojiPicker/commit/6f55afce3e6d8e0e3f1506305d729474ee5319f9
Yours: https://github.com/izyumkin/MCEmojiPicker/commit/ad839cbdddfb6ddb09dfd2936bd0b3c60b05d983

Mine: https://github.com/htmlprogrammist/EmojiPicker/commit/7592f8ab5b2c8b516aedb474d3cd0a353ec40d53
Yours: https://github.com/izyumkin/MCEmojiPicker/commit/e1219256cebb3aabdbae41703611e3f50abf73dc

Mine: https://github.com/htmlprogrammist/EmojiPicker/commit/31e8f3adf92fc1f0ec1b363d81731d82418a29e7
Yours: https://github.com/izyumkin/MCEmojiPicker/commit/29c9dd90f1e063b5955c26514fcfb44bf3eab16f

And of course it is commits only connected with the README, I don't count the rest of the changes showing nobility.

**Reply by htmlprogrammist** [reply]
it is much easier and more correct to use the suggestions I have presented earlier

**Reply by izyumkin** [reply]
Will these commits be enough to get you happy?

**Reply by htmlprogrammist** [reply]
Hello,
Of course not. If you still want to go your own way (not with the options I offered you when everything will be clear), then be kind enough to cherry pick at least the following commits:

https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/573f013d862e7a24667f9ba0791af3feb49e1ca5
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/442b1e5a019513f41c75b172c3b5cdff4b4fb63a
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/63b9e7ceca16acb1900ccfe74fa16bbf44d378ac
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/f42a836b9b282f947ccfd5407e58936b4069957d
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/8ffc465441110b9bfff0e6058cabfed10dd4f857
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/7c1eb0ee0fd5062d04d04d575aef9685b125717b
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/bd177aa3369c9e61355629808b21ab1cb08f4d40
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/5cf3c7159291ea5cb1dda3c33e119a7363226967
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/60a96a56c780127cda1562f301e5d371bc968d6a
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/7e37ea079cc4b1afc4977e20f8c36e2cdb06aa55
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/3dd164aa733b5e14ed7c04546e7d7bdbe7ad9ce3
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/5d0855312aba3287060c56dd6adb21eade0d1f6a
https://github.com/izyumkin/MCEmojiPicker/pull/7/commits/826a7167a12abe0bc629869765d0867383df92d2

**Reply by izyumkin** [reply]
Sorry mate, I've started adding commits, but what you're asking is unrealistic to do. I don't have the time or inclination to do it. 

In this [pull-request](https://github.com/izyumkin/MCEmojiPicker/pull/21) I chose a commit that changes the documentation, and it is your contribution that I want to include in my repository. People can see the rest of your work at your place, I left a [link](https://github.com/izyumkin/MCEmojiPicker/pull/21#issuecomment-1720857526) to it in the PR comments.

**Reply by htmlprogrammist** [reply]
You just threw away and disrespected everything I did. The main thing is that you took everything for yourself, but now you refuse to do anything for me. Okay, this will be on your conscience.

**Reply by izyumkin** [reply]
If you manage to include all the commits that you wrote above in one pull request, then I will gladly accept it. I just don't have time to do this, sorry.
